packages/terra-button/src/Button.jsx

```diff
@@ -115,221 +115,194 @@ const defaultProps = {
+  variant,
+  href,
+  onClick,
+  onMouseDown,
+  onBlur,
+  onFocus,
+  onKeyDown,
+  onKeyUp,
+  refCallback,
+  title,
+  ...customProps
+}) => {
+  const theme = React.useContext(ThemeContext);
+  const [focused, setFocused] = useState(false);
+  const [active, setActive] = useState(false);
+
+  useEffect(() => {
+    setActive(false);
+    if (isDisabled) {
+      setFocused(false);
     }
-    return null;
-  }
+  }, [isDisabled]);
 
-  handleOnBlur(event) {
-    this.setState({ focused: false });
+  const handleOnBlur = (event) => {
+    setFocused(false);
 
-    if (this.props.onBlur) {
-      this.props.onBlur(event);
+    if (onBlur) {
+      onBlur(event);
     }
-  }
+  };
 
-  handleClick(event) {
+  const handleClick = (event) => {
     // See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Button#Clicking_and_focus
     // Button on Firefox, Safari and IE running on OS X does not receive focus when clicked.
     // This will put focus on the button when clicked if it is not currently the active element.
     if (document.activeElement !== event.currentTarget) {
-      this.shouldShowFocus = false;
       event.currentTarget.focus();
-      this.shouldShowFocus = true;
     }
 
-    if (this.props.onClick) {
-      this.props.onClick(event);
+    if (onClick) {
+      onClick(event);
     }
-  }
+  };
 
-  handleKeyDown(event) {
+  const handleKeyDown = (event) => {
     // Add active state to FF browsers
     if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE) {
-      this.setState({ active: true });
+      setActive(true);
 
       // Follow href on space keydown when rendered as an anchor tag
-      if (this.props.href) {
-        // Prevent window scrolling
+      if (href) {
         event.preventDefault();
-        window.location.href = this.props.href;
+        window.location.href = href;
       }
     }
 
     // Add focus styles for keyboard navigation
     if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE || event.nativeEvent.keyCode === KeyCode.KEY_RETURN) {
-      this.setState({ focused: true });
+      setFocused(true);
     }
-
-    if (this.props.onKeyDown) {
-      this.props.onKeyDown(event);
+    if (onKeyDown) {
+      onKeyDown(event);
     }
-  }
+  };
 
-  handleKeyUp(event) {
+  const handleKeyUp = (event) => {
     // Remove active state from FF broswers
     if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE) {
-      this.setState({ active: false });
+      setActive(false);
     }
 
     // Apply focus styles for keyboard navigation
     if (event.nativeEvent.keyCode === KeyCode.KEY_TAB) {
-      this.setState({ focused: true });
+      setFocused(true);
     }
 
-    if (this.props.onKeyUp) {
-      this.props.onKeyUp(event);
+    if (onKeyUp) {
+      onKeyUp(event);
     }
-  }
+  };
 
-  handleFocus(event) {
-    if (this.shouldShowFocus) {
-      this.setState({ focused: true });
-    }
+  const handleFocus = (event) => {
+    setFocused(true);
+    event.currentTarget.focus();
```

Comment:
Why do we need to call the focus() method in the onFocus event handler?  Wouldn't the component already have focus here?